File: haskell/glue/src/Glue/Lib/Math/Trigonometric.hs
module Glue.Lib.Math.Trigonometric where

import Glue.Eval (Eval)
import Glue.Lib.Math.Trigonometric.Acos qualified as Acos
import Glue.Lib.Math.Trigonometric.Asin qualified as Asin
import Glue.Lib.Math.Trigonometric.Atan qualified as Atan
import Glue.Lib.Math.Trigonometric.Cos qualified as Cos
import Glue.Lib.Math.Trigonometric.Sin qualified as Sin
import Glue.Lib.Math.Trigonometric.Tan qualified as Tan
import Glue.Module (ModuleInfo, nativeModule)

trigonometric :: ModuleInfo Eval
trigonometric =
    nativeModule
        "ffi.math.trigonometric"
        [ ("sin", Sin.sin)
        , ("cos", Cos.cos)
        , ("tan", Tan.tan)
        , ("asin", Asin.asin)
        , ("acos", Acos.acos)
        , ("atan", Atan.atan)
        ]

File: dart/glue/lib/src/lib/math/trigonometric/trigonometric.dart
import 'package:glue/src/ir.dart';
import 'package:glue/src/module.dart';
import 'acos.dart';
import 'asin.dart';
import 'atan.dart';
import 'cos.dart';
import 'sin.dart';
import 'tan.dart';

/// Trigonometric module - trigonometric functions (radians)
/// Mirrors Haskell Glue.Lib.Math.Trigonometric exactly

/// The trigonometric module containing trigonometric functions
/// Mirrors Haskell Glue.Lib.Math.Trigonometric.trigonometric exactly
final ModuleInfo trigonometric = nativeModule('ffi.math.trigonometric', [
  // Direct trigonometric functions
  ('sin', IrNativeFunc(sin)),
  ('cos', IrNativeFunc(cos)),
  ('tan', IrNativeFunc(tan)),

  // Inverse trigonometric functions
  ('asin', IrNativeFunc(asin)),
  ('acos', IrNativeFunc(acos)),
  ('atan', IrNativeFunc(atan)),
]);
